src/Interpreters/LogicalExpressionsOptimizer.cpp

```diff
@@ -250,14 +247,11 @@ void LogicalExpressionsOptimizer::addInExpression(const DisjunctiveEqualityChain
         equals_expr_lhs = operands[0];
     }
 
-    auto tuple_function = std::make_shared<ASTFunction>();
-    tuple_function->name = "tuple";
-    tuple_function->arguments = value_list;
-    tuple_function->children.push_back(tuple_function->arguments);
+    auto tuple_literal = std::make_shared<ASTLiteral>(std::move(tuple));
```

Comment:
Why do you create `ASTLiteral`? If I'm not mistaken tuple is not a literal.

Reply:
Tuple can be literal, if it's all elements are literals. So, that's a problem, because we create tuple as function, format it in form of `(x1, x2, ..., x_n)` and send query to the remote shard, where this construction is parsed as literal, because all elements are literals.